TREnvironmentEditor/Parsing/EMConverter.cs

```diff
@@ -1,241 +1,158 @@
+            EMType.VerticalCollisionalPortal => JsonConvert.DeserializeObject<EMVerticalCollisionalPortalFunction>(jo.ToString(), this),
+            EMType.AdjustVisibilityPortal => JsonConvert.DeserializeObject<EMAdjustVisibilityPortalFunction>(jo.ToString(), this),
+            EMType.ReplaceCollisionalPortal => JsonConvert.DeserializeObject<EMReplaceCollisionalPortalFunction>(jo.ToString(), this),
+            EMType.RemoveCollisionalPortal => JsonConvert.DeserializeObject<EMRemoveCollisionalPortalFunction>(jo.ToString(), this),
+
+            // Sounds
+            EMType.AddSoundSource => JsonConvert.DeserializeObject<EMAddSoundSourceFunction>(jo.ToString(), this),
+            EMType.MoveSoundSource => JsonConvert.DeserializeObject<EMMoveSoundSourceFunction>(jo.ToString(), this),
+            EMType.RemoveSoundSource => JsonConvert.DeserializeObject<EMRemoveSoundSourceFunction>(jo.ToString(), this),
+
+            // Rooms
+            EMType.ModifyRoom => JsonConvert.DeserializeObject<EMModifyRoomFunction>(jo.ToString(), this),
+            EMType.ModifyOverlaps => JsonConvert.DeserializeObject<EMModifyOverlapsFunction>(jo.ToString(), this),
+            EMType.CopyRoom => JsonConvert.DeserializeObject<EMCopyRoomFunction>(jo.ToString(), this),
+            EMType.CopyVertexAttributes => JsonConvert.DeserializeObject<EMCopyVertexAttributesFunction>(jo.ToString(), this),
+            EMType.ImportRoom => JsonConvert.DeserializeObject<EMImportRoomFunction>(jo.ToString(), this),
+            EMType.CreateRoom => JsonConvert.DeserializeObject<EMCreateRoomFunction>(jo.ToString(), this),
+            EMType.CreateWall => JsonConvert.DeserializeObject<EMCreateWallFunction>(jo.ToString(), this),
+            EMType.GenerateLight => JsonConvert.DeserializeObject<EMGenerateLightFunction>(jo.ToString(), this),
+            EMType.MoveCamera => JsonConvert.DeserializeObject<EMMoveCameraFunction>(jo.ToString(), this),
+
+            // Models
+            EMType.ImportModel => JsonConvert.DeserializeObject<EMImportModelFunction>(jo.ToString(), this),
+            EMType.MirrorModel => JsonConvert.DeserializeObject<EMMirrorModelFunction>(jo.ToString(), this),
+            EMType.ConvertSpriteSequence => JsonConvert.DeserializeObject<EMConvertSpriteSequenceFunction>(jo.ToString(), this),
+            EMType.ConvertModel => JsonConvert.DeserializeObject<EMConvertModelFunction>(jo.ToString(), this),
+            EMType.ImportNonGraphicsModel => JsonConvert.DeserializeObject<EMImportNonGraphicsModelFunction>(jo.ToString(), this),
+            EMType.CopySpriteSequence => JsonConvert.DeserializeObject<EMCopySpriteSequenceFunction>(jo.ToString(), this),
+
+            // NOOP
+            EMType.NOOP => JsonConvert.DeserializeObject<EMPlaceholderFunction>(jo.ToString(), this),
+
+            _ => throw new InvalidOperationException(),
+        };
```

Comment:
Can this be coded in a more beautiful way, with the switch only mapping `T` instead of the entire result?

Something such as this

```c#
EMType type = …;
Type resultType = type switch {
    EMType.MirrorModel => EMImportModelFunction,
    …,
};
return (
    typeof(JsonConvert)
    .GetMethod("DeserializeObject")
    .MakeGenericMethod(resultType)
    .Invoke(jo.ToString(), this)
);
```
or something similar? (Maybe there's a compile-time friendly alternative to `GetMethod` in 2023, too)

Reply:
Sure, if there are plans to migrate to something else anyway, there's no use to waste time on reworking this approach.